gradio/route_utils.py

```diff
@@ -583,3 +599,64 @@ def starts_with_protocol(string: str) -> bool:
     """
     pattern = r"^[a-zA-Z][a-zA-Z0-9+\-.]*://"
     return re.match(pattern, string) is not None
+
+
+def delete_files_created_by_app(blocks: Blocks, age: int | None) -> None:
+    """Delete files that are older than age. If age is None, delete all files."""
+    print("age", age)
+    for temp_set in blocks.temp_file_sets:
+        to_remove = set()
+        for file in temp_set:
+            try:
+                file_path = Path(file)
+                modified_time = datetime.fromtimestamp(file_path.lstat().st_ctime)
+                print("modified_time", modified_time)
+                print("now", datetime.now())
+                print("age", (datetime.now() - modified_time).seconds)
+                if age is None or (datetime.now() - modified_time).seconds > age:
+                    print("Deleting file", file)
+                    os.remove(file)
+                    to_remove.add(file)
+            except FileNotFoundError:
+                continue
+        temp_set -= to_remove
```

Comment:
Why are we storing the unremoved files in `temp_set` here? Are we intending on doing something with these files later?

Reply:
You can't modify the contents of a set as you're iterating - you get an exception. So I'm removing from the set at the end.

Reply:
Ah got it